File: leetcode/q14.java
package leetcode;

public class q14 {
    class Solution {
        public String longestCommonPrefix(String[] strs) {
            int len = 0;
            char c;
            for(len = 0; len < strs[0].length(); len++) {
                c = strs[0].charAt(len);
                for(int i = 1; i < strs.length; i++) {
                    if(strs[i].length() == len) {
                        return strs[i];
                    }
                    if(strs[i].charAt(len) != c) {
                        return strs[0].substring(0, len);
                    }
                }
            }
    
            return strs[0];
        }
    }
}

File: top150/_01_ArrayString/q14.py
class Solution:
    def longestCommonPrefix(self, strs):
        strs.sort(key=lambda x: (len(x)))
        result = ""
        n = len(strs)
        for i, ch in enumerate(strs[0]):
            for j in range(1, n):
                if ch != strs[j][i]:
                    return result
            result += ch
        return result
